Gems/Atom/Feature/Common/Code/Source/Mesh/MeshFeatureProcessor.cpp

```diff
@@ -1026,6 +1022,118 @@ namespace AZ
             rayTracingFeatureProcessor->SetMesh(m_objectId, m_model->GetModelAsset()->GetId(), subMeshes);
         }
 
+        void ModelDataInstance::SetIrradianceData(RayTracingFeatureProcessor::SubMesh& subMesh,
+                const Data::Instance<RPI::Material> material, const Data::Instance<RPI::Image> baseColorImage)
+        {
+            RPI::MaterialPropertyIndex propertyIndex;
+
+            // Default to manual color, so we work properly with legacy materials that don't contain
+            // the irradianceColorSource enum, but have their old irradiance.color transformed to
+            // irradiance.manualColor through the 'versionUpdates' rules:
+            AZ::Name irradianceColorSource = AZ::Name("Manual");
```

Comment:
I think this comment isn't quite accurate. Since we are updating all the material types to include irradianceColorSource, all legacy _materials_ of those types will automatically have this property. What you are really doing here is allowing some other _material type_ besides the core ones to not define an irradianceColorSource property. 

```suggestion
            // Default to manual color, in case the material type doesn't have the concept of
            // manual vs. automatic irrandiance color.
            AZ::Name irradianceColorSource = AZ::Name("Manual");
```